src/driver/odoc_driver.ml

```diff
@@ -497,33 +497,56 @@ let render_stats env nprocs =
       in
       inner (0, 0, 0, 0, 0, 0, 0, 0))
 
-let run libs verbose odoc_dir html_dir stats nb_workers =
+let run libs verbose packages_dir odoc_dir odocl_dir html_dir stats nb_workers
+    odoc_bin voodoo package_name blessed dune_style =
```

Comment:
This `run` function is too complicated and it's unclear how the CLI should be used. To reduce maintenance and confusion, I suggest this is broken into sub-commands: `dune`, `opam` and `voodoo`.

Reply:
I though of that too.

However, in this case https://github.com/jonludlam/docs-ci/tree/next-gen should be updated. I suggest to do it in a later PR.

Reply:
Agree!

Reply:
Don't delay on requiring updates to the docs-ci branch, that's very much an experimental one and if it requires changes I'm happy to update it as required :-)

Reply:
and I'm definitely in favour of the sub-commands idea